### ISSUE TITLE
LIMIT issue with lib asynciterator

### ISSUE THREAD

**Issue by pfolz** [LinkedDataFragments/Client.js#17]
Hi,

With @gmontoya we tested your new iterator library with your ldfClient and we found an issue with the LIMIT operator.

An issue occurs when the size of the limit equal the size of the total number of results for this query.

**Ex**: consider a query that produces 5 results in total, if we add LIMIT 5 we will have the following error:
```
TypeError: Cannot read property '_closeWhenDone' of undefined
    at SparqlSelectIterator.destinationCloseWhenDone 
(/home/folz/test/delegateQuery/src/node_modules/Client.js-feature-asynciterator/node_modules/asynciterator/asynciterator.js:911:57)
```

But if we set the limit to 2 or 10 the error will not occurs.

We tested the following query, on http://fragments.dbpedia.org/2015/en:
```
SELECT * WHERE {
   <http://dbpedia.org/resource/Energy> <http://purl.org/dc/terms/subject> ?name
}
LIMIT 3
```

It works.

With LIMIT 1:
```
SELECT * WHERE {
   <http://dbpedia.org/resource/Energy> <http://purl.org/dc/terms/subject> ?name
}
LIMIT 1
```

The error occurs.

**Comment by RubenVerborgh** [comment]
Thanks for reporting, will investigate!

**Comment by RubenVerborgh** [comment]
I'm afraid I can't reproduce. I've tried:
```bash
$ bin/ldf-client http://fragments.dbpedia.org/2015/en 'SELECT * WHERE { <http://dbpedia.org/resource/Energy> <http://purl.org/dc/terms/subject> ?name } LIMIT 1'
$ bin/ldf-client http://fragments.dbpedia.org/2015/en 'SELECT * WHERE { <http://dbpedia.org/resource/Energy> <http://purl.org/dc/terms/subject> ?name } LIMIT 3'
$ bin/ldf-client http://fragments.dbpedia.org/2015/en 'SELECT * WHERE { <http://dbpedia.org/resource/Energy> <http://purl.org/dc/terms/subject> ?name } LIMIT 5'
```
but all of them succeed consistently.

Do you have another use case?

**Comment by RubenVerborgh** [comment]
Oh wait, I was working with a slightly newer version of AsyncIterator. Apparently [this seemingly unrelated commit](https://github.com/RubenVerborgh/AsyncIterator/commit/8ddcb30fd445377967d1f29cda2c2943d3dc5daf) fixed the issue. I will release a new version that includes this commit.

**Comment by RubenVerborgh** [comment]
Fixed in [AsyncIterator 1.0.1](https://github.com/RubenVerborgh/AsyncIterator/releases/tag/v1.0.1).
